gitlab/__init__.py

```diff
@@ -210,6 +211,55 @@ def from_config(cls, gitlab_id=None, config_files=None):
             user_agent=config.user_agent,
         )
 
+    @classmethod
+    def merge_config(
+        cls,
+        options,
+        gitlab_id=None,
+        config_files=None,
+    ):
+        """Create a Gitlab connection by merging configuration in the following
+        order:
```

Comment:
s/order/priority/

I'm assuming that arguments via the CLI arguments have the highest priority and will not be over-written by other configs.